packages/core/core/src/RequestTracker.js

```diff
@@ -764,15 +764,19 @@ export class RequestGraph extends ContentGraph<
           let fileNameNodeId = this.getNodeIdByContentKey(
             'file_name:' + basename,
           );
+
           // Find potential file nodes to be invalidated if this file name pattern matches
-          let above = this.getNodeIdsConnectedTo(
+          let above: Array<FileNode> = [];
+          for (const nodeId of this.getNodeIdsConnectedTo(
```

Comment:
I originally had this code using `.filter` but Flow didn't like that, it didn't know that `above` only contained file nodes when passed in to `invalidateFileNameNode` on line 783.